tests/functional/test_basic.py

```diff
@@ -585,3 +585,15 @@ def test_several_scripts2(pyi_builder_spec):
     Verify each script has it's own global vars (basic test).
     """
     pyi_builder_spec.test_spec('several-scripts2.spec')
+
+
+@pytest.mark.win32
+def test_pe_checksum(pyi_builder):
+    import pefile
+
+    pyi_builder.test_source("print('hello')")
+    exes = pyi_builder._find_executables('test_source')
+    assert exes
+    for exe in exes:
+        pe = pefile.PE(exe)
+        assert pe.verify_checksum()
```

Comment:
Would it make sense to test the checksum via "official" `IMAGEHLP.DLL` implementation instead of `pefile`'s one?

```python
import ctypes
from ctypes import wintypes

headersum  = ctypes.wintypes.DWORD()
checksum = ctypes.wintypes.DWORD()
ctypes.windll.imagehlp.MapFileAndCheckSumA(
    ctypes.c_char_p(filename.encode('utf8')), 
    ctypes.byref(headersum), 
    ctypes.byref(checksum))
    
assert headersum.value == checksum.value
```

Reply:
yeah why not

Reply:
What makes you think that the encoding should be utf-8?

Reply:
Good point. Maybe the W variant of the function call should be used instead?

Reply:
Which simplifies the call in the above example even further:
```
ctypes.windll.imagehlp.MapFileAndCheckSumW(
    ctypes.c_wchar_p(filename), 
    ctypes.byref(headersum), 
    ctypes.byref(checksum))
```

Reply:
There we go. Weird 16 bit unicode is now ctypes's problem.